packages/contracts-bedrock/tasks/validate-spacers.ts

```diff
@@ -0,0 +1,37 @@
+import { task } from 'hardhat/config'
+
+task(
+  'validate-spacers',
+  'validates that spacer variables are in the correct positions'
+).setAction(async (args, hre) => {
+  const names = await hre.artifacts.getAllFullyQualifiedNames()
+  for (const name of names) {
+    // Script is remarkably slow because of getBuildInfo, so better to skip test files since they
+    // don't matter for this check.
+    if (name.includes('.t.sol')) {
+      continue
+    }
+
+    const buildInfo = await hre.artifacts.getBuildInfo(name)
+    for (const source of Object.values(buildInfo.output.contracts)) {
+      for (const [contractName, contract] of Object.entries(source)) {
+        const storageLayout = (contract as any).storageLayout
+        for (const variable of storageLayout?.storage || []) {
+          if (variable.label.startsWith('spacer_')) {
+            const [, slot, offset] = variable.label.split('_')
+            if (slot !== variable.slot) {
+              throw new Error(
+                `${contractName} ${variable.label} is in slot ${variable.slot} but should be in ${slot}`
+              )
+            }
+            if (parseInt(offset, 10) !== variable.offset) {
+              throw new Error(
+                `${contractName} ${variable.label} is at offset ${variable.offset} but should be at ${offset}`
+              )
```

Comment:
I noticed that you're not checking that the length given in the space name matches the type...
which could be done easily enough given all spacers are currently either a mapping, uint or address.

That would be a slight improvement, but I'm fine either way. Will approve now and you can resolve the comment if you just want to merge it.

Reply:
Oops, good catch. Added an simple version of this to the script.